packages/cli/src/commands/generate/auth/auth.js

```diff
@@ -55,13 +55,28 @@ const addWebRender = (content, authProvider) => {
   const redwoodProviderLines = redwoodProvider.split('\n').map((line) => {
     return '  ' + line
   })
+  const customRenderOpen = authProvider.render
+    ? authProvider.render.reduce(
+        (acc, component) => acc + indent + `<${component}>`,
+        ''
+      )
+    : ''
+  const customRenderClose = authProvider.render
+    ? authProvider.render.reduce(
+        (acc, component) => indent + `</${component}>` + acc,
+        ''
+      )
+    : ''
+
   const renderContent =
+    customRenderOpen +
     indent +
     `<AuthProvider client={${authProvider.client}} type="${authProvider.type}">` +
```

Comment:
This will generate `<AuthProvider client=etherum type="etherum">`, right?
But `etherum` is undefined (it will be injected later). So the day we switch this file over to TS things will break. Not 100% sure what the best way forward is here is, but the two viable options I see are
 1. make `client` optional on `<AuthProvider>` and don't generate it if `authProvider.client` is `falsey`
 2. Provide a dummy etherum client that we know will be replaced by the injector

Reply:
I'm not sure I follow, and am not a TS pro. 

Are you saying that TS will throw for a null `ethereum` since I am using `React.cloneElement` to inject it?

Would it be solved by adding a `let ethereum` before the `ApolloInjector`?

```js
// new line
let ethereum = null;

const ApolloInjector = ({ children }) => {
...
  return React.cloneElement(children, { client: ethereum })
}

ReactDOM.render(
  <FatalErrorBoundary page={FatalErrorPage}>
    <FetchConfigProvider>
      <ApolloInjector>
        <AuthProvider client={ethereum} type="ethereum">
          <RedwoodProvider>
...
)
```

Reply:
Adding `let ethereum = null` like that is a step towards my second option. However, if you look here https://github.com/redwoodjs/redwood/blob/6cf2a9a38c73dbf82d7ec1c75479eec3f7234143/packages/auth/src/AuthProvider.tsx#L64 you see that `client` has to be of type `SupportedAuthClients`, and `null` is not included in that type. 

But maybe this is the best option for now. And then when we make this a TS file we can just type-cast null to `Etherium`, add a comment, and everyone's happy 🙂 

The other option I mentioned is to have just `<AuthProvider type="ethereum">` but then we would have to make sure we handle an `undefined` client in AuthProvider. 

Personally I'd just go with setting ethereum to null as you suggested.

Reply:
OK I think I understand now. So the story goes like this:

1. TS is impatient and says "I _glanced over this file_ and the react element `AuthProvider` needs `ethereum` of type `SupportedAuthClients`, but `ethereum` is null. Throw an error yo!"
2. `React.cloneElement` comes in to save the day but its already too late. Everyone is sad.

If we go with that, then I don't think I need to make any more changes in this PR

Reply:
Haha, yeah, that's a great way to put it :)  

I had a second look at the supabase auth implementation yesterday, which has a very similar decoder to the one you've implemented. Found a potential issue, but need to do some more investigation. I'll get back to you when I know more.

Reply:
My hunch was correct. There is an issue with the supabase auth. I wrote up an issue about it, and a PR to fix it. #1652 #1653

Reply:
And sorry it took so long to get back to you 🙁